core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala

```diff
@@ -339,7 +339,11 @@ class FunctionPullingContainerProxy(
 
     // wait for container creation when cold start
     case Event(ClientCreationCompleted(proxy), _: NonexistentData) =>
-      self ! ClientCreationCompleted(proxy.orElse(Some(sender())))
+      akka.pattern.after(3.milliseconds, actorSystem.scheduler) {
```

Comment:
Even if it generally does not happen in production, it would be great to add a small delay to avoid the cycle.